src/main.js

```diff
@@ -0,0 +1,178 @@
+/**
+ *    Copyright 2022, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+import { logger } from 'log';
+import { Cookies, SetCookie } from 'cookies';
+import {
+  createInstance,
+  enums as OptimizelyEnums,
+} from '@optimizely/optimizely-sdk/dist/optimizely.lite.es';
+
+import { getDatafile, dispatchEvent } from './optimizely_helper';
+
+const AKAMAI_CLIENT_ENGINE = "javascript-sdk/akamai-edgeworker";
+const COOKIE_NAME_OPTIMIZELY_VISITOR_ID = 'optimizely_visitor_id';
+const VARIABLE_NAME_USER_ID = "PMUSER_OPTIMIZELY_USER_ID";
+const VARIABLE_NAME_DECISION_EVENT = "PMUSER_OPTIMIZELY_DECISION_EVENT"; 
+
+const SHOULD_DISPATCH_EVENT = false;
+
+let logStash = [];
+
+// Generates a random 6 digit user id for demo purpose. For production uses, please use user Ids from your system
+// or autogenerate a uuid.
+function generateRandomUserId() {
+  return `${Math.floor(Math.random() * 899999 + 100000)}`;
```

Comment:
We'll want to update this to use `uuid` as well, right?

Reply:
I added the custom version of uuid that we discussed earlier. The uuid library does not seem to work with akamai because it relies on some methods from crypto which is not supported on akamai edge workers.